trunk/source/WitnessParser/src/de/uni_freiburg/informatik/ultimate/witnessparser/yaml/WitnessEntry.java

```diff
@@ -0,0 +1,35 @@
+package de.uni_freiburg.informatik.ultimate.witnessparser.yaml;
+
+/**
+ * Represents a generic witness type in a YAML-based witness file.
+ *
+ * @author Manuel Bentele
+ */
+public abstract class WitnessEntry implements IYamlProvider {
+
+	private final String mName;
+	protected final Metadata mMetadata;
+
+	public WitnessEntry(final String name, final Metadata metadata) {
+		mName = name;
```

Comment:
if you base hashcode on that, either ensure that mName is never null or handle the case in hashCode()